### ISSUE TITLE
dotenv state diverge from .env file

### ISSUE THREAD

**Issue by ayoubedd** [cachix/devenv#1591]
**Describe the bug**
am using devenv with direnv integration.
when updating `.env` devenv doesn't rebuild.

**To reproduce**
create a dummy `.env` with a couple of envs.
enable `dotenv`.
```nix
  ...
  dotenv.enable = true;
  ...
```
build shell, update `.env`. then devenv is out of sync.
not sure if its a devenv problem or a direnv problem.

**Version**
devenv 1.3.1

**Comment by domenkozar** [comment]
@sandydoo huh, shouldn't that be detected via cache paths?

**Comment by ayoubedd** [comment]
@sandydoo yes and no, when i reload it registers the change, but i have to do it every time i change .env file

**Comment by ayoubedd** [comment]
@sandydoo apparently when i created a project recently but i used the flake template to instantiate devenv.

```sh
nix flake init --template github:cachix/devenv
``

it works just fine, it picks change to .env just fine

**Comment by alanscodelog** [comment]
I'm running into this problem I think (or possibly #1363 ) but not even `direnv reload` updates the env... I have to delete `.devenv/nix-eval-cache.db` (i tried deleting stuff that was not /state until something worked). `devenv shell` works but I don't want to be dropped into a new shell. I had a flakes setup before and it worked, but had to switch to regular devenv because I needed containers.

**Comment by justinTM** [comment]
would be cool to see this fixed if it is indeed a bug!

**Comment by domenkozar** [comment]
> I'm running into this problem I think (or possibly [#1363](https://github.com/cachix/devenv/issues/1363) ) but not even `direnv reload` updates the env... I have to delete `.devenv/nix-eval-cache.db` (i tried deleting stuff that was not /state until something worked). `devenv shell` works but I don't want to be dropped into a new shell. I had a flakes setup before and it worked, but had to switch to regular devenv because I needed containers.

What version of devenv?

**Comment by alanscodelog** [comment]
@domenkozar 1.7.0  I'll try with 1.8. Didn't see there was a newer version.

**Comment by alanscodelog** [comment]
Updated to 1.8.1, it's still happening.
